Error on quarkus:dev after upgrade to 1.10.2.Final only with multiple persistence units

After upgrading my project to quarkus 1.10.2.Final. I got an exception while running `mvn quarkus:dev` 
I'm sure the issue because multiple persistence units configured in my project.

**StackTrace**
```
 java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
        [error]: Build step io.quarkus.deployment.dev.HotDeploymentConfigFileBuildStep#setupConfigFileHotDeployment threw an exception: java.lang.IllegalStateException: Duplicate key true
        at java.util.stream.Collectors.lambda$throwingMerger$0(Collectors.java:133)
        at java.util.HashMap.merge(HashMap.java:1254)
        at java.util.stream.Collectors.lambda$toMap$58(Collectors.java:1320)
        at java.util.stream.ReduceOps$3ReducingSink.accept(ReduceOps.java:169)
        at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1382)
        at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:482)
        at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:472)
        at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
        at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
        at io.quarkus.deployment.dev.HotDeploymentConfigFileBuildStep.setupConfigFileHotDeployment(HotDeploymentConfigFileBuildStep.java:19)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:972)
        at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
        at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
        at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2046)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1578)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
        at java.lang.Thread.run(Thread.java:748)
        at org.jboss.threads.JBossThread.run(JBossThread.java:479)

        at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:254)
        at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:135)
        at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:55)
        at io.quarkus.deployment.dev.IsolatedDevModeMain.firstStart(IsolatedDevModeMain.java:75)
        at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:361)
        at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:52)
        at io.quarkus.bootstrap.app.CuratedApplication.runInCl(CuratedApplication.java:129)
        at io.quarkus.bootstrap.app.CuratedApplication.runInAugmentClassLoader(CuratedApplication.java:82)
        at io.quarkus.deployment.dev.DevModeMain.start(DevModeMain.java:144)
        at io.quarkus.deployment.dev.DevModeMain.main(DevModeMain.java:63)
Caused by: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
        [error]: Build step io.quarkus.deployment.dev.HotDeploymentConfigFileBuildStep#setupConfigFileHotDeployment threw an exception: java.lang.IllegalStateException: Duplicate key true
        at java.util.stream.Collectors.lambda$throwingMerger$0(Collectors.java:133)
        at java.util.HashMap.merge(HashMap.java:1254)
        at java.util.stream.Collectors.lambda$toMap$58(Collectors.java:1320)
        at java.util.stream.ReduceOps$3ReducingSink.accept(ReduceOps.java:169)
        at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1382)
        at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:482)
        at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:472)
        at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
        at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
        at io.quarkus.deployment.dev.HotDeploymentConfigFileBuildStep.setupConfigFileHotDeployment(HotDeploymentConfigFileBuildStep.java:19)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:972)
        at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
        at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
        at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2046)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1578)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
        at java.lang.Thread.run(Thread.java:748)
        at org.jboss.threads.JBossThread.run(JBossThread.java:479)

        at io.quarkus.builder.Execution.run(Execution.java:116)
        at io.quarkus.builder.BuildExecutionBuilder.execute(BuildExecutionBuilder.java:79)
        at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:148)
        at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:252)
        ... 9 more
Caused by: java.lang.IllegalStateException: Duplicate key true
        at java.util.stream.Collectors.lambda$throwingMerger$0(Collectors.java:133)
        at java.util.HashMap.merge(HashMap.java:1254)
        at java.util.stream.Collectors.lambda$toMap$58(Collectors.java:1320)
        at java.util.stream.ReduceOps$3ReducingSink.accept(ReduceOps.java:169)
        at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1382)
        at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:482)
        at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:472)
        at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
        at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
        at io.quarkus.deployment.dev.HotDeploymentConfigFileBuildStep.setupConfigFileHotDeployment(HotDeploymentConfigFileBuildStep.java:19)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:972)
        at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
        at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
        at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2046)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1578)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
        at java.lang.Thread.run(Thread.java:748)
        at org.jboss.threads.JBossThread.run(JBossThread.java:479)

```



**To Reproduce**
I can reproduce the issue easily by defining multiple persistence units in a new standalone project.

**Configuration**
```properties

quarkus.datasource.db-kind=mysql
quarkus.datasource.jdbc.url=jdbc:mysql://localhost:3306/xyz?useSSL=false&allowPublicKeyRetrieval=true
quarkus.datasource.username=root
quarkus.datasource.password=test123


# Quartz datasource
quarkus.datasource.quartz.db-kind=mysql
quarkus.datasource.quartz.jdbc.url=jdbc:mysql://localhost:3306/abc?useSSL=false&allowPublicKeyRetrieval=true&serverTimezone=UTC
quarkus.datasource.quartz.username=root
quarkus.datasource.quartz.password=test123

quarkus.hibernate-orm.database.generation=none
quarkus.hibernate-orm.packages=com.test.model

quarkus.hibernate-orm."system".datasource=quartz
quarkus.hibernate-orm."system".packages=com
```

--- a/core/deployment/src/main/java/io/quarkus/deployment/dev/HotDeploymentConfigFileBuildStep.java
+++ b/core/deployment/src/main/java/io/quarkus/deployment/dev/HotDeploymentConfigFileBuildStep.java
@@ -17,7 +17,8 @@ public class HotDeploymentConfigFileBuildStep {
         if (processor != null) {
             Map<String, Boolean> watchedFilePaths = files.stream()
                     .collect(Collectors.toMap(HotDeploymentWatchedFileBuildItem::getLocation,
-                            HotDeploymentWatchedFileBuildItem::isRestartNeeded));
+                            HotDeploymentWatchedFileBuildItem::isRestartNeeded,
+                            (isRestartNeeded1, isRestartNeeded2) -> isRestartNeeded1 || isRestartNeeded2));
             processor.setWatchedFilePaths(watchedFilePaths);
         }
         return null;

--- a/extensions/hibernate-orm/deployment/src/test/java/io/quarkus/hibernate/orm/multiplepersistenceunits/MultiplePersistenceUnitsImportSqlHotReloadScriptTest.java
+++ b/extensions/hibernate-orm/deployment/src/test/java/io/quarkus/hibernate/orm/multiplepersistenceunits/MultiplePersistenceUnitsImportSqlHotReloadScriptTest.java
@@ -1,0 +1,53 @@
+package io.quarkus.hibernate.orm.multiplepersistenceunits;
+
+import static org.hamcrest.Matchers.is;
+
+import java.util.function.Function;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.hibernate.orm.multiplepersistenceunits.model.annotation.inventory.Plane;
+import io.quarkus.hibernate.orm.multiplepersistenceunits.model.annotation.shared.SharedEntity;
+import io.quarkus.hibernate.orm.multiplepersistenceunits.model.annotation.user.User;
+import io.quarkus.hibernate.orm.multiplepersistenceunits.model.annotation.user.subpackage.OtherUserInSubPackage;
+import io.quarkus.test.QuarkusDevModeTest;
+import io.restassured.RestAssured;
+
+/**
+ * See https://github.com/quarkusio/quarkus/issues/13722
+ */
+public class MultiplePersistenceUnitsImportSqlHotReloadScriptTest {
+    @RegisterExtension
+    final static QuarkusDevModeTest TEST = new QuarkusDevModeTest()
+            .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class)
+                    .addPackage(Plane.class.getPackage().getName())
+                    .addPackage(SharedEntity.class.getPackage().getName())
+                    .addPackage(User.class.getPackage().getName())
+                    .addPackage(OtherUserInSubPackage.class.getPackage().getName())
+                    .addClass(MultiplePersistenceUnitsSqlLoadScriptTestResource.class)
+                    .addAsResource("application-multiple-persistence-units-annotations.properties", "application.properties")
+                    .addAsResource("import-sharedentity.sql", "import.sql"));
+
+    @Test
+    public void testImportSqlScriptHotReload() {
+        String expectedName = "import.sql load script entity";
+        assertBodyIs(expectedName);
+
+        String hotReloadExpectedName = "modified import.sql load script entity";
+        TEST.modifyResourceFile("import.sql", new Function<String, String>() {
+            @Override
+            public String apply(String s) {
+                return s.replace(expectedName, hotReloadExpectedName);
+            }
+        });
+        assertBodyIs(hotReloadExpectedName);
+    }
+
+    private void assertBodyIs(String expectedBody) {
+        RestAssured.when().get("/multiple-persistence-units/orm-sql-load-script/2").then().body(is(expectedBody));
+    }
+
+}

--- a/extensions/hibernate-orm/deployment/src/test/java/io/quarkus/hibernate/orm/multiplepersistenceunits/MultiplePersistenceUnitsSqlLoadScriptTestResource.java
+++ b/extensions/hibernate-orm/deployment/src/test/java/io/quarkus/hibernate/orm/multiplepersistenceunits/MultiplePersistenceUnitsSqlLoadScriptTestResource.java
@@ -1,0 +1,32 @@
+package io.quarkus.hibernate.orm.multiplepersistenceunits;
+
+import javax.inject.Inject;
+import javax.persistence.EntityManager;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import io.quarkus.hibernate.orm.multiplepersistenceunits.model.annotation.shared.SharedEntity;
+
+@Path("/multiple-persistence-units/orm-sql-load-script")
+public class MultiplePersistenceUnitsSqlLoadScriptTestResource {
+
+    public static final String NO_ENTITY_MESSAGE = "no entity";
+
+    @Inject
+    EntityManager em;
+
+    @GET
+    @Path("/{id}")
+    @Produces(MediaType.TEXT_PLAIN)
+    public String getName(@PathParam("id") long id) {
+        SharedEntity entity = em.find(SharedEntity.class, id);
+        if (entity != null) {
+            return entity.getName();
+        }
+
+        return NO_ENTITY_MESSAGE;
+    }
+}

--- a/extensions/hibernate-orm/deployment/src/test/resources/import-sharedentity.sql
+++ b/extensions/hibernate-orm/deployment/src/test/resources/import-sharedentity.sql
@@ -1,0 +1,2 @@
+INSERT INTO SharedEntity(id, name) VALUES(1, 'default sql load script entity');
+INSERT INTO SharedEntity(id, name) VALUES(2, 'import.sql load script entity');

